pkg/hubble/parser/threefour/parser.go

```diff
@@ -459,6 +461,19 @@ func decodeTraceReason(tn *monitor.TraceNotify) pb.TraceReason {
 	}
 }
 
+func decodeIpTraceId(tn *monitor.TraceNotify) *pb.IPTraceID {
+	if tn == nil {
+		return &pb.IPTraceID{
+			TraceId:      0,
+			IpOptionType: 0,
+		}
+	}
+	return &pb.IPTraceID{
+		TraceId:      tn.GetIPTraceID(),
+		IpOptionType: uint32(option.Config.EnableIPOptionTracing),
+	}
```

Comment:
return `nil` when possible, that way the json encoder will skip the property and emit `{flow:{…}}` instead of an empty trace id `{flow:{…, ip_trace_id:{}}}`
```suggestion
	if tn == nil || option.Config.EnableIPOptionTracing == 0 {
		return nil
	}
	return &pb.IPTraceID{
		TraceId:      tn.GetIPTraceID(),
		IpOptionType: uint32(option.Config.EnableIPOptionTracing),
	}
```